### PR TITLE
Add Summer School 2021 overview page

### LINKED ISSUE #2153: Create new Overview page for Summer School

### PR DESCRIPTION
**Pull request by techtolentino**
Fixes #2153 and #2093

- adds Summer school 2021 Overview page
- adds all images (compressed)

### CODE REVIEW

File: components/textbook-beta/SummerSchoolCoursePages.vue
Comment (korgan00) [inline components/textbook-beta/SummerSchoolCoursePages.vue:312]:
```suggestion
    margin: 0 0 0.5rem;
```
typo?

**Review by korgan00 — CHANGES_REQUESTED** [review]
Good job!

File: components/textbook-beta/RelatedMaterialSection.vue
Comment (techtolentino) [reply components/textbook-beta/RelatedMaterialSection.vue:37]: I agree @korgan00 - there's a lot of opportunity for refactors. I can create issues for each component that we should abstract/refactor.

But since this task was supposed to be a `size - small` — I'm leaving out a lot of the refactoring that I noticed, was needed.

Once all the reviews are done, I'll include the refactor issues here

File: components/textbook-beta/SummerSchoolCoursePages.vue
Comment (techtolentino) [reply components/textbook-beta/SummerSchoolCoursePages.vue:66]: Similar to above, since this was a small task, the refactoring was out of scope... I created issues here:

https://github.com/Qiskit/qiskit.org/issues/2168
https://github.com/Qiskit/qiskit.org/issues/2167

File: components/textbook-beta/PrerequisiteMaterialSection.vue
Comment (eddybrando) [inline components/textbook-beta/PrerequisiteMaterialSection.vue:78]: Unnecessary use of the [repeat function](https://developer.mozilla.org/en-US/docs/Web/CSS/repeat()):

```suggestion
      grid-template-columns: 1fr;
```

File: components/textbook-beta/RelatedMaterialSection.vue
Comment (eddybrando) [inline components/textbook-beta/RelatedMaterialSection.vue:98]: Unnecessary use of the [repeat function](https://developer.mozilla.org/en-US/docs/Web/CSS/repeat()):

```suggestion
      grid-template-columns: 1fr;
```

File: components/textbook-beta/PrerequisiteMaterialSection.vue
Comment (eddybrando) [inline components/textbook-beta/PrerequisiteMaterialSection.vue:70]: Nice catch! `gap` is now [widely supported](https://caniuse.com/flexbox-gap). 🙂

File: components/textbook-beta/SummerSchoolCoursePages.vue
Comment (eddybrando) [reply components/textbook-beta/SummerSchoolCoursePages.vue:19]: @korgan00, could you please elaborate a little bit on this?

**Review by eddybrando — APPROVED** [review]
Very nice job @techtolentino! Thanks a lot! 🙂 

Please take a look at my comments before merging. Most are in the code itself. I'll just add one:

* Since no changes were purposefully made to the dependencies, I think that the changes to the **package-lock.json** shouldn't be included.

File: components/textbook-beta/RelatedMaterialSection.vue
Comment (y4izus) [inline components/textbook-beta/RelatedMaterialSection.vue:1]: Have you checked if you can reuse the `PrerequisiteMaterialSection` component instead of creating this new `RelatedMaterialSection` component?

File: components/textbook-beta/SummerSchoolCoursePages.vue
Comment (y4izus) [inline components/textbook-beta/SummerSchoolCoursePages.vue:1]: Have you checked if you can reuse the `CoursePagesSection` component instead of creating this `SummerSchoolCoursePagesSection`?

**Review by y4izus — CHANGES_REQUESTED** [review]
Thank you Randy! I have a couple of comments to see if we can clean the code, and other one related with the styles:
- The 2 first ones are on the code 👇 
- Related with the styles: we have [this issue](https://github.com/Qiskit/qiskit.org/issues/2093), and I think it's the moment to fix it. Look how it looks on this page:
Captura de pantalla 2021-09-20 a las 14 17 42

File: components/textbook-beta/RelatedMaterialSection.vue
Comment (techtolentino) [reply components/textbook-beta/RelatedMaterialSection.vue:1]: I started to turn this into a reusable component but with this task being a `S`, adding refactors into this felt out of scope... so I created an issue to handle extending the component https://github.com/Qiskit/qiskit.org/issues/2167

File: components/textbook-beta/SummerSchoolCoursePages.vue
Comment (techtolentino) [reply components/textbook-beta/SummerSchoolCoursePages.vue:1]: Same here, 

I started to turn this into a reusable component but with this task being a `S`, adding refactors into this felt out of scope... so I created an issue to handle extending the component https://github.com/Qiskit/qiskit.org/issues/2168

**Comment by techtolentino** [comment]
@y4izus (updated w/ comments, and included a fix for https://github.com/Qiskit/qiskit.org/issues/2093)

File: components/textbook-beta/RelatedMaterialSection.vue
Comment (y4izus) [reply components/textbook-beta/RelatedMaterialSection.vue:1]: ok, I see your point. For this issue I'm ok with this, but I think we should reuse the components and clean code before merge the feature branch. What do you think, @techtolentino , @eddybrando , @korgan00 ?

File: components/textbook-beta/SummerSchoolCoursePages.vue
Comment (korgan00) [reply components/textbook-beta/SummerSchoolCoursePages.vue:19]: if you use the index in the `selectCourse`, something like `selectCourse(courseIndex)`.
Then `activeCourse` that currently is...
```
  get activeCourse () {
    const activeCourse = this.courses.find(
      ({ label }) => label === this.activeCourseLabel
    )
    return activeCourse || null
  }
```

... could be something like...

```
  get activeCourse () {
    return this.courses[this. activeCourseIndex] || null
  }
```

Also, it could prevent some future errors if two lectures have the same label.

File: components/textbook-beta/RelatedMaterialSection.vue
Comment (korgan00) [reply components/textbook-beta/RelatedMaterialSection.vue:1]: It is fine for me.

File: constants/appLinks.ts
Comment (y4izus) [inline constants/appLinks.ts:67]: I think that it's better to have this `introductionCoursePrerequisites` on the `introduction-course` page

File: constants/appLinks.ts
Comment (y4izus) [inline constants/appLinks.ts:90]: I think that it's better to have this `summerSchool2021Prerequisites` on the `quantum-computing-and-quantum-learning-2021` page.

Also, if in the future we have repeated prerequisites, we could think in a file for having them and not repeat code

File: constants/appLinks.ts
Comment (y4izus) [inline constants/appLinks.ts:60]: This interface already exists on `PrerequisiteMaterialSection`, you don't need to create a new one

**Review by y4izus — CHANGES_REQUESTED** [review]
Thank you Randy! Just some comments for reorganizing code

File: pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue
Comment (y4izus) [inline pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue:54]: this is tab or is my impression?

File: pages/textbook-beta/course/introduction-course.vue
Comment (y4izus) [inline pages/textbook-beta/course/introduction-course.vue:152]: this doesn't have the type of the variable `Prerequisite[]`

File: pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue
Comment (y4izus) [inline pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue:33]: you already have this interface on the other page. It could be a nice idea to move it to another file and export it from there. Maybe you can follow the same you did with the `Prerequisite` interface until we have a better solution

File: pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue
Comment (y4izus) [inline pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue:56]: You need to add the type here `GeneralLink`

File: pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue
Comment (y4izus) [inline pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue:134]: If we are not going to need this, you can remove it

File: pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue
Comment (y4izus) [inline pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue:244]: type `Prerequisite[]` missing

**Review by y4izus — CHANGES_REQUESTED** [review]
Thank you Randy!! I think this is my last round of requests :), could you take a look at them?

File: pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue
Comment (techtolentino) [reply pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue]: fixed

File: pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue
Comment (techtolentino) [reply pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue]: Great suggestion, thanks!

File: pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue
Comment (techtolentino) [reply pages/textbook-beta/summer-school/quantum-computing-and-quantum-learning-2021.vue]: removed!

File: components/textbook-beta/SummerSchoolCoursePages.vue
Comment (techtolentino) [reply components/textbook-beta/SummerSchoolCoursePages.vue]: @korgan00 - I didn't ignore this suggestion, I agree actually. We just merged to unblock future work that depends on this PR... 

Just wanted to comment to acknowledge your suggestion 👍🏽

**Pull request merged**
